### PR TITLE
Batch table tests

### PR DESCRIPTION
**Pull request by lilleyse**
For #3177 

Added tests for `Cesium3DTileBatchTableResources`. I also cleaned up an unrelated test for `ModelInstanceCollection`.

### CODE REVIEW

File: Source/Scene/Cesium3DTileBatchTableResources.js [hunk 1]
```diff
@@ -593,7 +593,7 @@ define([
                 }
             };
 
-            return combine(uniformMap, batchUniformMap);
+            return combine(batchUniformMap, uniformMap);
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileBatchTableResources.js]: The reason I swapped these is `Model` gives `pickUniformMapLoaded` the uniform map generated in `uniformMapLoaded`, and since some uniforms have the same name `combine` defaulted to the callbacks set in `uniformMapLoaded`. With the order switched, it uses the pick callbacks instead, helping code coverage.

Comment (pjcozzi) [inline Source/Scene/Cesium3DTileBatchTableResources.js]: Can you add a comment about this here?

File: Specs/Scene/Cesium3DTileBatchTableResourcesSpec.js [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+
+    afterAll(function() {
+        scene.destroyForSpecs();
+    });
+
+    afterEach(function() {
+        scene.primitives.removeAll();
+    });
+
+    function verifyRender(tileset) {
```
Comment (pjcozzi) [inline Specs/Scene/Cesium3DTileBatchTableResourcesSpec.js]: Do we have enough use cases at this point so we are ready to make some common functions in a separate PR?

File: Specs/Scene/Cesium3DTileBatchTableResourcesSpec.js [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+    var centerLongitude = -1.31995;
+    var centerLatitude = 0.69871;
+
+    var withBatchTableUrl = './Data/Cesium3DTiles/Batched/BatchedWithBatchTable/';
+    var withoutBatchTableUrl = './Data/Cesium3DTiles/Batched/BatchedWithoutBatchTable/';
+    var batchLengthZeroUrl = './Data/Cesium3DTiles/Batched/BatchedNoBuildings/';
+
+    var result = new Color();
+
+    var fakeContentProvider = {
```
Comment (pjcozzi) [inline Specs/Scene/Cesium3DTileBatchTableResourcesSpec.js]: Name this `mock` instead of `fake`.  https://github.com/AnalyticalGraphicsInc/cesium/blob/testing-guide/Documentation/Contributors/TestingGuide/README.md#mocks

File: Specs/Scene/Cesium3DTileBatchTableResourcesSpec.js (hunk 2 above)
Comment (lilleyse) [inline Specs/Scene/Cesium3DTileBatchTableResourcesSpec.js]: Yes that will be the next one.

**Comment by pjcozzi** [comment]
Just those trivial comments.  This looks good!

**Comment by lilleyse** [comment]
Updated.

**Comment by pjcozzi** [comment]
Great!
